internal/controller/object/object.go

```diff
@@ -291,27 +334,29 @@ func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalUpdate{}, err
 	}
 
-	meta.AddAnnotations(obj, map[string]string{
-		v1.LastAppliedConfigAnnotation: string(cr.Spec.ForProvider.Manifest.Raw),
-	})
+	if e.applyExtractor == nil {
```

Comment:
perhaps we could think about two external client implementation, to avoid using `applyExtractor` as a flag, or to make the current `external` client more configurable by introducing some hooks?